BookingSystem/Frontend/src/hotelSearch/containers/HotelSearch.js

```diff
@@ -1,29 +1,132 @@
 import React, { Component } from "react";
 import { connect } from "react-redux";
 import { bindActionCreators } from "redux";
-import { HotelSearchActions } from "../actions";
+import { HotelSearchActions, hotelSearchActionType } from "../actions";
 import HotelSearch from "../components/HotelSearch";
 
 const mapStateToProps = state => {
-  const hotelSearch = state.hotelSearch;
+  const hotels = state.hotelSearch.hotels;
+  const form = state.hotelSearch.searchForm;
   return {
-    hotels: hotelSearch.hotels,
-    isFetching: hotelSearch.isFetching,
-    error: hotelSearch.error
+    hotels: hotels.hotels,
+    isFetching: hotels.isFetching,
+    error: hotels.error,
+    currentCity: form.currentCity,
+    currentCountry: form.currentCountry,
+    countryOptions: form.countryOptions,
+    cityOptions: form.cityOptions,
+    search: state.router.location.pathname
   };
 };
 
+class HotelSearchContainer extends Component {
+  state = {
+    startDate: new Date(),
+    endDate: new Date(),
+    roomSize: 0
+  };
+
+  componentDidMount() {
+    this.props.getHotels({});
+  }
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const data = {
+      ...this.state,
+      cityId: this.props.currentCity.cityId,
+      countryId: this.props.currentCountry.countryId
+    };
+    this.props.getHotels(data);
+  };
+
+  handleReset = event => {
+    this.props.getHotels({});
+  };
+
+  handleInputChange = event => {
+    const target = event.target;
+    this.setState({
+      [target.name]: target.value
+    });
+  };
+
+  handleStartDateChange = date => {
+    this.setState({
+      startDate: date
+    });
+  };
+
+  handleEndDateChange = date => {
+    this.setState({
+      endDate: date
+    });
+  };
+
+  handleCountryOptionsChange = search => {
+    if (search === "") return;
+    this.props.loadCountryOptions({ countryName: search });
+  };
+
+  handleCityOptionsChange = search => {
+    if (search === "") return;
+    const countryName = this.props.currentCountry.name;
+    this.props.loadCityOptions({
+      cityName: search,
+      countryName: countryName
+    });
+  };
+
+  handleCountryChange = data => {
+    this.props.setCountry({ countryId: data.value, name: data.label });
+  };
+
+  handleCityChange = data => {
+    this.props.setCity({ cityId: data.value, name: data.label });
+  };
+
+  render() {
+    return (
+      <HotelSearch
+        startDate={this.state.startDate}
+        endDate={this.state.endDate}
+        roomSize={this.state.roomSize}
+        city={this.props.currentCity}
```

Comment:
why are startDate, endDate and roomSize parts of State, but the rest of search options are in props?